sparkplug-core/src/main/scala/springnz/sparkplug/core/MapPimpers.scala

```diff
@@ -16,5 +16,17 @@ object MapPimpers {
           case _            ⇒ throw new Exception // should never happen - just there to suppress warning
         }
 
+    def getFirst(seqA: A*): Option[B] = seqA.length match {
+      case 0 ⇒ None
+      case _ ⇒ map.get(seqA.head) match {
+        case None  ⇒ getFirst(seqA.tail: _*)
+        case value ⇒ value
+      }
+
+    }
```

Comment:
Slightly better looking with `seqA.headOption`. Although it will still potentially overflow the stack 
```
    def getFirst(seqA: A*): Option[B] =
      seqA.headOption flatMap { head ⇒
        map.get(head) match {
          case None  ⇒ getFirst(seqA.tail: _*)
          case value ⇒ value
        }
      }
```